apiserver/tools.go

```diff
@@ -49,12 +51,18 @@ func (h *toolsDownloadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		tarball, err := h.processGet(r, st)
 		if err != nil {
 			logger.Errorf("GET(%s) failed: %v", r.URL, err)
-			sendError(w, errors.NewBadRequest(err, ""))
+			if err := sendError(w, errors.NewBadRequest(err, "")); err != nil {
```

Comment:
For the same reason I thought your other suggestion was good, no, I don't think this is a good idea. I made this change because I couldn't readily tell whether or not we were dealing with failures to send errors correctly.

Errors handling should be lazy; i.e., it should be handled at the last possible moment. This has a few benefits:
- The errors stack gets built up as it gets traced back to where it was called from.
- This gives the caller flexibility in how it's handled and centralizes that handling.
- Making this change percolated throughout the code revealing new error paths; I think it's much clearer now what might fail.